dpdk/vr_dpdk_interface.c

```diff
@@ -927,13 +937,34 @@ dpdk_if_tx(struct vr_interface *vif, struct vr_packet *pkt)
     rte_pktmbuf_dump(stdout, m, 0x60);
 #endif
 
+    vr_stats = vif_get_stats(tx_queue->q_vif, lcore_id);
     if (likely(tx_queue->txq_ops.f_tx != NULL)) {
         tx_queue->txq_ops.f_tx(tx_queue->q_queue_h, m);
         if (lcore_id == VR_DPDK_PACKET_LCORE_ID)
             tx_queue->txq_ops.f_flush(tx_queue->q_queue_h);
+
+        /**
+         * Update counters for:
+         *  - packets enqueued to the interface successfully.
+         *  - packets which have been dropped when .f_tx() could not send.
+         *  If we write to ring instead of NIC's queue, count it as a ring
+         *  enqueue.
+         */
+        if (likely(tx_queue->txq_ops.f_stats != NULL)) {
+            tx_queue->txq_ops.f_stats(tx_queue->q_queue_h, &port_stats, 0);
+
+            if (tx_queue->txq_ops.f_tx == rte_port_ring_writer_ops.f_tx) {
+                vr_stats->vis_iftxrngenqpkts = port_stats.n_pkts_in;
```

Comment:
n_pkts_out would be better (instead of n_pkts_in), but it looks like this is what the DPDK patch has...